### ISSUE TITLE
[HOLD for payment 2023-09-08] [$1000] Split  money- participants detail hidden after clicking on show more button

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#24476]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1 Split money with multiple users
2 On the second page of split money, click on show more.
3 Verify that participants are hidden which can be scrolled.
Split money
4 Click on the split money box > this will open the split money details.
5 Click on show more
## Expected Result:
 I was expecting that the entire detail page would be scrollable when I clicked on the show more button. Reason being, participants' detail & amount are the most important components. But things like date and merchant are fixed rather than participant details.
## Actual Result:
Gives the illusion that participants are hidden or deleted. No scrollbar. Sometimes, it's hard to scroll.
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.53-1
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/38435837/da9b5932-0063-4881-9ff0-bd522478a685

https://github.com/Expensify/App/assets/38435837/e40aab9b-78f0-4063-81ec-d44b150446b6

**Expensify/Expensify Issue URL:**
**Issue reported by:** @avi-shek-jha
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1691097988530299

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01eb2b30669469492f</li>
        <li>Upwork Job ID: 1692405727765413888</li>
        <li>Last Price Increase: 2023-08-25</li>
<li>Automatic offers: </li>
<ul>
<li>DanutGavrus | Contributor | 26385876</li>
<li>avi-shek-jha | Reporter | 26385878</li>
</ul></li>
    </ul>
</details>

**Comment by jeet-dhandha** [comment]
It's intentional 👍, not a mistake. We added inline scrolling to address a problem: when more than 8 people are chosen, the main description gets hidden, and users have to scroll down to see it.

**Comment by jeet-dhandha** [comment]
@jliexpensify - This might be a feature request but one way to solve this is to show 3 person and for all person we can add a "View All" / "View All Participants" / "All Participants" - button, which will open up a participant list including all the selected members. What are your thoughts on this one ?

**Comment by jliexpensify** [comment]
Hmm yeah, I agree - it's a little annoying, but I don't see this as a huge deal.

> for all person we can add a "View All" / "View All Participants" / "All Participants" - button, which will open up a participant list including all the selected members

That's an interesting idea - @jeet-dhandha could you propose that in the Open Source channel? For now, I'll close off this GH and if we decide to move forward, a new one will be opened.

**Comment by DanutGavrus** [comment]
Hi, @jeet-dhandha and @jliexpensify! On smaller devices, we have a problem, participant details are **not** *appearing at all*, so you *don't* have the option to scroll:  

https://github.com/Expensify/App/assets/56603839/26743177-db11-4142-8b40-ffa99bb83492

The bug was reported [here](https://app.slack.com/client/T03SC9DTT/C049HHMV9SM/thread/C049HHMV9SM-1691087177.963409) on Slack, but the reporter stated to use web zoom as he didn't have a small device. We are using `ScrollViews` for our pages, thus content on smaller devices becomes scrollable(*Share code page* from Setting is an example), but the `ScrollView` is missing from the *Request money page*. We also have: *"If a new page is added, I verified it's using the ScrollView component to make it scrollable when more elements are added to the page."* in the [contributing guides](https://github.com/Expensify/App/blob/main/contributingGuides/REVIEWER_CHECKLIST.md).

As stated by @jeet-dhandha, it is intentional to scroll the participants list, and I think we can achieve that by wrapping the *Money request page* in a `ScrollView`, and then, setting a **max height** for the current participants `ScrollView` to the total of showing 3 participants before having to scroll.

Do you think this is a valid bug? Should we reopen this one and fix it here, should we open that reported bug from Slack and fix it there, or is it not a bug? Thank you both for your time!

**Comment by jliexpensify** [comment]
Original [issue](https://expensify.slack.com/archives/C049HHMV9SM/p1691087177963409) - @mallenexpensify what do you think? Should we consider this one as a bug?

@DanutGavrus just curious, what is defined as a `smaller device`?

**Comment by DanutGavrus** [comment]
> just curious, what is defined as a smaller device?

The above video has a resolution of `930px width × 588px height` for example.

**Comment by jliexpensify** [comment]
Getting a second opinion internally - https://expensify.slack.com/archives/C01SKUP7QR0/p1692252620222139

**Comment by jliexpensify** [comment]
Hi @eVoloshchak - do you mind checking out @DanutGavrus 's proposal [here](https://github.com/Expensify/App/issues/24476#issuecomment-1676874283)? It seems like this is a bug that we would want to look at and possibly fix, thanks!

**Comment by jliexpensify** [comment]
Not overdue, waiting on C+

**Comment by eVoloshchak** [comment]
@DanutGavrus's [proposal](https://github.com/Expensify/App/issues/24476#issuecomment-1676874283) looks good to me!
I think we need to do items 1-4, while 5 is a separate issue, we already have a reviewer checklist check to prevent pages without the ScrollView

🎀👀🎀 C+ reviewed!

**Comment by jliexpensify** [comment]
Just awaiting @Gonals to assign a Contributor.

**Comment by Gonals** [comment]
All yours, @DanutGavrus

**Comment by DanutGavrus** [comment]
@eVoloshchak
PR ready for review: https://github.com/Expensify/App/pull/26179

This one has proven to be a little bit more challenging than sounded. I've done as many exhaustive tests as I possibly could, on all platforms, sizes & orientations(as partly seen in the PR videos).

**Comment by jliexpensify** [comment]
Payment summary:

- Bug Reporter: @avi-shek-jha $250
- Contributor: @DanutGavrus $1000 + $500 (speed bonus)
- C+: @eVoloshchak $1000 + $500 (speed bonus) - request on New.Dot

[Upwork job](https://www.upwork.com/jobs/~01eb2b30669469492f)

Is that correct?

@eVoloshchak please complete the checklist! @DanutGavrus please accept the offer.

**Comment by eVoloshchak** [comment]
- The PR that introduced the bug has been identified. Link to the PR: I can't pinpoint the PR that introduced this bug, the screen didn't have the ScrollView initially (which is the root cause), but this was surfaced when 'Show more' button was added
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: this discussion has already happened before, which resulted in the following item being added to the reviewer checklist: `If a new page is added, I verified it's using the ScrollView component to make it scrollable when more elements are added to the page`
- Determine if we should create a regression test for this bug.
_Is it easy to test for this bug? Yes
Is the bug related to an important user flow? Yes
Is it an impactful bug? No_
This is a visual bug that can be reproduced on a small number of devices (it has to be a fairly small device/window) that doesn't prevent the user from continuing to use the app. I think the checklist item is enough, the problem won't occur for all of the newly created pages, I don't think a regression test is needed here

**Comment by eVoloshchak** [comment]
> Contributor: @DanutGavrus $1000 + $500 (speed bonus)
C+: @eVoloshchak $1000 + $500 (speed bonus) - request on New.Dot

> Is that correct?

Looks like has introduced a regression in https://github.com/Expensify/App/issues/26544
This was quickly resolved by a follow-up [PR](https://github.com/Expensify/App/pull/26655) from @DanutGavrus, it was deployed to production 4 days ago, so the payment date isn't due just yet (payments will be issued 7 days after all regressions are fixed (ie: deployed to production), [source](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#regressions))

**Comment by DanutGavrus** [comment]
We also have this issue https://github.com/Expensify/App/issues/26418 which was not opened to payment and was mentioned in my follow up PR. In the dupe issue, which was opened to external contributors but was fixed by me, it was suggested that we [only pay the reporter](https://github.com/Expensify/App/issues/26544#issuecomment-1712007377). You can see in my follow-up [PR changes](https://github.com/Expensify/App/pull/26655/files) that I only forgot to add 1 prop to a ScrollView and caused a very small issue(the Next button's position was not docked to bottom anymore, but everything was still functional). The original [PR](https://github.com/Expensify/App/pull/26179/files) was a little bit tricky(as you can see with the double ScrollView) and my focus was on the important things(as you can see from the attached videos and the variety of tests) and I've also voluntarily fixed another issue in advance which was not reported yet(added a ScrollView for our custom keyboard page too). IMO, the button position seems a very small detail which could easily be omitted by anyone. Those being said, I would like to ask if this small detail may not be considered a regression for the previously mentioned things, and if we may continue with the payment as is. Thanks!

**Comment by jliexpensify** [comment]
Thanks for the context @DanutGavrus - since I am not an Engineer, I will ask @Gonals to step in and evaluate your [comment](https://github.com/Expensify/App/issues/24476#issuecomment-1712867139) and make the final call.

If he agrees with this comment:

>IMO, the button position seems a very small detail which could easily be omitted by anyone. 

And we choose to not count this as a regression, then the payouts in this [comment](https://github.com/Expensify/App/issues/24476#issuecomment-1707652076) would still be valid. If it is a regression, then we should apply a 50% penalty (i.e. $750 all up).

**Comment by jliexpensify** [comment]
Heya @Gonals - can I get your thoughts [here](https://github.com/Expensify/App/issues/24476#issuecomment-1712975557)?

**Comment by Gonals** [comment]
I think that is a fair assessment. It is a small issue in a moderately complex PR. I think, in this particular case, we could go ahead with payment as usual.

**Comment by jliexpensify** [comment]
Great, thanks Alberto. Paying out as per this [comment](https://github.com/Expensify/App/issues/24476#issuecomment-1707652076) as the second PR has been deployed for 7 days.

**Comment by jliexpensify** [comment]
Paid and job closed, cheers!

**Comment by JmillsExpensify** [comment]
$1,500 payment approved for @eVoloshchak based on BZ summary.
